packages/utils/sets-utils/src/cell-set-utils.js

```diff
@@ -509,30 +509,44 @@ export function treeToCellPolygonsBySetNames(
  * Given a tree with state, get the sizes of the
  * sets currently marked as "visible".
  * @param {object} currTree A tree object.
+ * @param {array} allNamePaths Array of all paths.
  * @param {array} selectedNamePaths Array of arrays of strings,
- * representing set "paths".
+ * representing selected paths.
  * @param {object[]} setColor Array of objects with the
  * properties `path` and `color`.
  * @param {string} theme "light" or "dark" for the vitessce theme
  * @returns {object[]} Array of objects
  * with the properties `name`, `size`, `key`,
  * and `color`.
  */
-export function treeToSetSizesBySetNames(currTree, selectedNamePaths, setColor, theme) {
+export function treeToSetSizesBySetNames(
+  currTree, allNamePaths, selectedNamePaths, setColor, theme,
+) {
   const sizes = [];
-  selectedNamePaths.forEach((setNamePath) => {
-    const node = treeFindNodeByNamePath(currTree, setNamePath);
+
+  const contains = (path, paths) => paths.some((p) => {
```

Comment:
Can you please add a comment about what this code is doing?

Reply:
I just added. Is it clear enough?

Reply:
Yes, thanks!